common/src/api/external/mod.rs

```diff
@@ -2692,6 +2770,74 @@ pub struct SwitchPortBgpPeerConfig {
     /// The address of the peer.
     pub addr: IpAddr,
 }
+*/
+
+/// A BGP peer configuration for an interface. Includes the set of announcements
+/// that will be advertised to the peer identified by `addr`. The `bgp_config`
+/// parameter is a reference to global BGP parameters. The `interface_name`
+/// indicates what interface the peer should be contacted on.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+pub struct BgpPeer {
```

Comment:
This is not net-new, it's moved from `nexus/external_api/params.rs` to be used as a common data structure for PUT/POST and GET requests.